src/local/watcher.js

```diff
@@ -263,53 +255,47 @@ class LocalWatcher {
   // It can be a file moved out. So, we wait a bit to see if a file with the
   // same checksum is added and, if not, we declare this file as deleted.
   onUnlinkFile (filePath) {
-    const clear = () => {
-      clearTimeout(this.pending[filePath].timeout)
-      delete this.pending[filePath]
+    let timeout
+    const stopChecking = () => {
+      clearTimeout(timeout)
     }
-    const done = () => {
-      clear()
+    const execute = () => {
+      this.pending.clear(filePath)
       log.info(`${filePath}: File deleted`)
       this.prep.deleteFile(this.side, {path: filePath}, this.done)
     }
     const check = () => {
       if (this.checksums === 0) {
-        done()
+        execute()
       } else {
-        this.pending[filePath].timeout = setTimeout(check, 100)
+        timeout = setTimeout(check, 100)
       }
     }
-    this.pending[filePath] = {
-      clear,
-      done,
-      check,
-      timeout: setTimeout(check, 1250)
-    }
+    this.pending.add(filePath, {stopChecking, execute})
+    timeout = setTimeout(check, 1250)
   }
 
   // Folder deletion detected
   //
   // We don't want to delete a folder before files inside it. So we wait a bit
   // after chokidar event to declare the folder as deleted.
   onUnlinkDir (folderPath) {
-    const clear = () => {
-      clearInterval(this.pending[folderPath].interval)
-      delete this.pending[folderPath]
+    let interval
+    const stopChecking = () => {
+      clearInterval(interval)
     }
-    const done = () => {
-      clear()
+    const execute = () => {
+      this.pending.clear(folderPath)
       log.info(`${folderPath}: Folder deleted`)
       this.prep.deleteFolder(this.side, {path: folderPath}, this.done)
     }
     const check = () => {
-      if (!this.hasPendingChild(folderPath)) { done() }
-    }
-    this.pending[folderPath] = {
-      clear,
-      done,
-      check,
-      interval: setInterval(check, 350)
+      if (!this.pending.hasPendingChild(folderPath)) {
+        execute()
+      }
     }
+    this.pending.add(folderPath, {stopChecking, execute})
```

Comment:
For `stopChecking`, it's the pending map that removes `folderPath`. But for `execute`, it is the `execute` function that removes it. I think it could be clearer to remove it from the same class for both cases. I don't mind if it is the `stopChecking` and `execute` functions, or in the `pendingMap`, but I'd prefer something consistent for both use cases.